src/OpenTelemetry.Exporter.Geneva/Metrics/GenevaMetricExporter.cs

```diff
@@ -885,6 +929,31 @@ private static void SerializeHistogramBucketWithTLV(in HistogramBucket bucket, b
         MetricSerializer.SerializeUInt32(buffer, ref bufferIndex, Convert.ToUInt32(bucket.BucketCount));
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private void GetCustomAccountAndNamespaceIfAny(in ReadOnlyTagCollection tags, out string monitoringAccount, out string metricNamespace)
+    {
+        monitoringAccount = this.defaultMonitoringAccount;
+        metricNamespace = this.defaultMetricNamespace;
+
+        foreach (var tag in tags)
```

Comment:
we are iterating several times over tags. Check if we can avoid it.

Reply:
Fixed in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1111/commits/85d03d139815a64ee8278607bc91be091d822e0f.